src/tabs/tab-header-bar.tsx

```diff
@@ -254,13 +261,20 @@ export function TabHeaderBar({
     function isElementRegistered(element: HTMLElement) {
       return navigationAPI.current?.isRegistered(element) ?? false;
     }
-    function isElementDisabled(element: HTMLElement) {
+    function isElementFocusable(element: HTMLElement) {
       if (element instanceof HTMLButtonElement) {
-        return element.disabled || element.closest(`.${styles['tabs-tab-disabled']}`);
+        return !(
+          element.disabled ||
+          (element.closest(`.${styles['tabs-tab-disabled']}`) && !element.closest(`.${styles['tabs-tab-focusable']}`))
```

Comment:
Would it make sense to attach the disabled style whenever the element is disabled regardless of whether it has a reason or not? Then it should be enough to only check styles and do not check the disabled attribute.

Reply:
The tab can be disabled when there is no reason, but it can have `tabs-tab-disabled` in both cases. Yet, it should only have `tabs-tab-focusable` when it is either not disabled or disabled with disabled reason.

Reply:
Which actually means that it should be enough to use `tabs-tab-focusable` if it is only present when the tab is actually focusable no matter why.

Reply:
Made it even simpler 

```javascript
return !element.disabled || element.closest(focusableTabSelector);
```